service/src/main/java/uk/gov/hmcts/reform/fpl/model/ChildParty.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmcts.reform.fpl.model;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import uk.gov.hmcts.reform.fpl.enums.PartyType;
+import uk.gov.hmcts.reform.fpl.model.common.EmailAddress;
+import uk.gov.hmcts.reform.fpl.model.common.Party;
+import uk.gov.hmcts.reform.fpl.model.common.Telephone;
+
+import java.util.Date;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public final class ChildParty extends Party {
+    private final String gender;
+    private final String genderIdentification;
+    private final String livingSituation;
+    private final String livingSituationDetails;
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
+    private final Date addressChangeDate;
+    private final String keyDates;
+    private final String careAndContactPlan;
+    private final String adoption;
+    private final String placementOrderApplication;
+    private final String placementCourt;
+    private final String mothersName;
+    private final String fathersName;
+    private final String fathersResponsibility;
+    private final String socialWorkerName;
+    private final Telephone socialWorkerTelephoneNumber;
+    private final String additionalNeeds;
+    private final String additionalNeedsDetails;
+    private final String detailsHidden;
+    private final String detailsHiddenReason;
+    private final String litigationIssues;
+    private final String litigationIssuesDetails;
+
+    @Builder(toBuilder = true)
+    public ChildParty(String partyId,
```

Comment:
Is this one required? Sorry not sure what's the power of Lombok - actually didn't use it in the past!

Reply:
yes! 

` public List<Element<Child>> addHiddenValues(CaseData caseData) {
        return caseData.getChildren1().stream()
            .map(element -> {
                Child.ChildBuilder childBuilder = Child.builder();

                if (element.getValue().getParty().getPartyId() == null) {
                    childBuilder
                        .party(element.getValue().getParty().toBuilder()
                            .partyId(UUID.randomUUID().toString())
                            .partyType(PartyType.INDIVIDUAL)
                            .build());
                } else {
                    childBuilder.party(element.getValue().getParty().toBuilder().build());
                }

                return Element.<Child>builder()
                    .id(element.getId())
                    .value(childBuilder.build())
                    .build();
            })
            .collect(toList());
    }`

we use in the above method. It sets all our values to a builder, and lets us add our hidden values so they are persisted into the db :)